File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int, int> table;
	    int mx = 0;
	    for (auto&& n : nums) table[n]++, mx = max(mx, table[n]);
	    vector<vector<int>> res(mx);
	    for (auto&& r : res) {
		    vector<int> unique;
		    for (auto&& [num, amount] : table) {
			    if (amount > 0) unique.push_back(num);
			    amount--;
		    }
		    r = unique;
	    }
	    return res;
    }
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
	def findMatrix(self, nums: List[int]) -> List[List[int]]:

		result = [[nums[0]]]

		for num in nums[1:]:

			check = True
			for row in range(len(result)):

				if num not in result[row]:
					result[row].append(num)
					check = False
					break

			if check == True:
				result.append([num])

		return result